test/functional/plugin/lsp/testutil.lua

```diff
@@ -39,8 +39,12 @@ M.create_server_definition = [[
         })
         local handler = handlers[method]
         if handler then
-          local response, err = handler(method, params)
-          callback(err, response)
+          if debug.getinfo(handler, 'u').nparams == 3 then
```

Comment:
Seems like this isn't working correctly in puc-lua.

Reply:
I updated all the handlers to be async.

Reply:
Not using such a hacky solution sounds great. But changing all the handlers to be async might be too much. I haven't tested this much in my environments so it might not be accurate yet, but how about something like:

```diff
        local handler = handlers[method]
+       local delay = (opts.delay or {})[method] ---@type integer? (ms)
+       if delay then -- simulate 'slow', asynchronous response from LSP
+         local original_callback = callback
+         callback = function(err, response)
+           vim.defer_fn(function(),
+             original_callback(err, response)
+           end, delay)
+         end
+       end

        if handler then
          local response, err = handler(method, params)
          callback(err, response)
```

so one can write like and avoid touching other handlers:
```lua
        local server = _create_server {
          delay = {
            ['initialize'] = 100, -- simulate LSP server is responding 100ms late
          }
        }
```

Reply:
I actually did exactly this at first. But passing a callback is less code, provides more control, and is easier to understand.